modules/library/render/src/main/java/org/geotools/renderer/lite/StreamingRenderer.java

```diff
@@ -2557,7 +2557,7 @@ private void processSymbolizers(final Graphics2D graphics,
                     // we need to preserve the topology if we end up applying buffer for perp. offset
                     boolean preserveTopology = style instanceof LineStyle2D && ((LineStyle2D) style).getPerpendicularOffset() != 0 &&
                             (source instanceof Polygon || source instanceof MultiPolygon);
-                    Geometry g = clip(shape.getGeometry(), clipper, preserveTopology);
```

Comment:
I have a question about this.  I originally just thought it was weird terminology, but I think its a bug.
If you look you'll see the "boolean preserveTopology".  However, if you look at the original code (clip() function in streaming renderer, removed in this pull request) you'll see that it actually flips this boolean (line 2611).

I expect the flip is incorrect, but not sure.

NOTE: if you don't flip it, then it uses the non-robust clip() which doesnt fail.

If flipping is correct, then I'll change it to;
 Geometry g = clipper.clipFailResistant(shape.getGeometry(),  !preserveTopology,  1);

Or, better;
 boolean dontPreserveTopology = ...
Geometry g = clipper.clipFailResistant(shape.getGeometry(),  !dontPreserveTopology,  1);